src/inbox/mod_inbox_rdbms.erl

```diff
@@ -216,6 +242,33 @@ check_result({updated, Res}, Exp) ->
 check_result(Result, _) ->
     {error, {bad_result, Result}}.
 
+check_result({selected, []}) ->
+    ok;
+
+check_result({selected, [{Val}]}) ->
+    case Val of
```

Comment:
May be matched in function head or even better - in separate function (may be reused in next clause).